File: ruby/scraper.rb
require 'nokogiri'
require 'httparty'
require 'csv'

class Scraper
  def initialize(url)
    @url = url
  end

  def perform
    page = HTTParty.get(@url)
    parsed_page = Nokogiri::HTML(page)
    data = parsed_page.css('h3').map(&:text)
    CSV.open('google_search_results.csv', 'w') { |csv| csv << data }

    puts data
  end
end

Scraper.new('https://www.google.com/search?q=scraping').perform

File: python/scraper.py
from bs4 import BeautifulSoup
import scrapy

class Scraper(scrapy.Spider):
    name = 'SimpleGoogleScraper'
    start_urls = ['https://www.google.com/search?q=scraping']

    def parse(self, response):
      soup = BeautifulSoup(response.text, 'lxml')
      headers = soup.find_all('h3')
      yield {
        "url": response.url,
        "headers": [h.get_text() for h in headers],
      } 
      # for next_page in response.css('a#pnnext'):
      #     yield response.follow(next_page, self.parse)
